onnxruntime/core/providers/js/allocator.cc

```diff
@@ -10,15 +10,21 @@ namespace onnxruntime {
 namespace js {
 
 void* JsCustomAllocator::Alloc(size_t size) {
+  if (size == 0) {
```

Comment:
guess that is in line with c malloc.

Reply:
Ideally an allocator should not handle Alloc(0) and Free(nullptr). But somehow Tensor class calls Free(nullptr) so need to add a condition to handle that. So I also add support for Alloc(0) to make a set.